scripts/simulateFunctionsJavaScript.js

```diff
@@ -1,12 +1,37 @@
-const { simulateRequest, getDecodedResultLog } = require("../FunctionsSandboxLibrary")
-const requestConfig = require("../Functions-request-config.js")
+const { simulateRequest, getDecodedResultLog, getRequestConfig } = require("../FunctionsSandboxLibrary")
 
-;(async () => {
+const runSimulation = async (requestConfig) => {
   const { resultLog, result, success } = await simulateRequest(requestConfig)
 
   console.log(`\n${resultLog}`)
   if (success) {
-    console.log(`Value returned from souce code: ${result}\n${getDecodedResultLog(requestConfig, result)}`)
+    console.log(`Value returned from source code: ${result}\n${getDecodedResultLog(requestConfig, result)}`)
     return
   }
+}
+
+(async () => {
+    const unvalidatedRequestConfig = require("../Functions-request-config.js")
+    const requestConfig = getRequestConfig(unvalidatedRequestConfig)
+
+    if (requestConfig.secretsLocation === 1) {
```

Comment:
replace "1" with "Remote"

Reply:
Else, it is going to be a larger change to make Functions-request-config.js have separate exports as opposed to a default export.

Reply:
So, its whatever we prefer.  Either repeat code for the Location object "enum", create a new utils file that will hold `Location`, or leave it as is.

Reply:
IMO, best solution is to address this in a separate PR that introduces a `utils` file that also holds things like the gas overrides.

Reply:
This "enum" object will likely go away entirely when we have proper TypeScript implementation for the plugin.

Reply:
Yes, let's move to common.js in a separate PR.